src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -83,10 +84,16 @@ public Instance(LinkedHashMap<String, Object> instanceMap, LinkedHashMap<String,
             // a jmxtree that is not completely initialized and would be missing some attributes
         }
 
-        this.minCollectionPeriod = (Integer) instanceMap.get("min_collection_interval");
+        this.minCollectionPeriod = (Integer) this.instanceMap.get("min_collection_interval");
```

Comment:
Not really your fault, but we have so much room for `NullPointerException` in this constructor.....

Reply:
Indeed, though the same thing while working on it. Do you think we should wrap these in try/catch blocks?

Reply:
What about:
```
    private boolean emptyDefaultHostname;

...

        try {
            this.emptyDefaultHostname = (Boolean) this.instanceMap.get("empty_default_hostname");
        } catch (NullPointerException e) {
            this.emptyDefaultHostname = false;
        }
```